src/app/services/rider.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http'; 
+import { Observable } from 'rxjs';
+import { User } from '../models/user.model';
+
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RiderService {
+
+  constructor(private httpClient: HttpClient) { }
+
+  private url: string = "http://localhost:8080/users/"; 
+  private body: string;
+  private httpOptions = {
+    headers: new HttpHeaders({"Content-Type": "application/json"}), 
+    observe: "response" as "body"
+  };
+
+  showAllDrivers(): Observable<User> {
```

Comment:
There exists in other pull requests a UserService; perhaps this should be included in a UserService as well, as it deals with "user" records. And for a method which should show all drivers, this method will be showing all drivers and riders. Take a look at how the UserController narrows the results based and change your request url based on that.